### DISCUSSION TITLE
Google Cloud Nightscout still on 14.2.6 despite update

### DISCUSSION THREAD

**Discussion by emp-00** [NightscoutFoundation/xDrip#3444] [Q&A] [answered]
Dear Team, I have updated my Google Cloud Nightscout via "Update platform" as per instructions (last update was half a year ago). After that I rebooted the server VM twice via the menu option "Reboot Server (Nightscout)" but my NS remains at 14.2.6. 

I saw that 15.0.0. is already available since Oct 18, 2023 - this is a major stable release, right? Am I doing anything wrong or is this release just not yet part of the Google Clound Nightscout package?

Here's my status page:
![image](https://github.com/NightscoutFoundation/xDrip/assets/60714349/0ea21925-3e45-41e8-a6a1-59c4d804a2dd)

**Comment by Navid200** [comment]
Please run "Install Nightscout Phase 1".
Then, reboot the server and then, see if it fixes is.

**Reply by emp-00** [reply]
Thanks for helping - I did as you proposed. Result - no change, still running NS 14.2.6 - new status see below... Does your package already include NS 15.X or maybe it's still on 14.2.6 ?

![image](https://github.com/NightscoutFoundation/xDrip/assets/60714349/f2fbd924-7305-4a50-8689-1a237992f177)
![image](https://github.com/NightscoutFoundation/xDrip/assets/60714349/a7c1d552-8618-407c-bdb5-134d3bc26ea9)
![image](https://github.com/NightscoutFoundation/xDrip/assets/60714349/b9ece3cb-9b3f-4db5-a5a0-c8fd3ee76130)

**Reply by Navid200** [reply]
No, an update will update to 15.

Run "Bootstrap the stable release" from the Google Cloud setup submenu.
After it is completed and goes through the forced reboot, after it comes back up, run phase 1 again.

That should update to 15 for sure.

**Reply by emp-00** [reply]
Understood. However, when selecting "Bootrap stable release" I get a warning message that "my installation will be modified". This is a very vague statement - let's be concrete: Will I lose all my customized NS settings incl. plugins and the BG history database as well? Can I avoid a complete re-configuration?

**Reply by Navid200** [reply]
If you created your setup a very long time ago, the local version of our repository is located on your virtual machine in a directory that needed to be moved to evolve our setup.  That move allowed us to make our setup more flexible.  Running bootstrap does the move.
The move does not move where the Nightscout variables reside.  Neither does it move where the database resides.  But, let's be as cautious as we can be.  

Create a backup and move it off your virtual machine to your computer.  That has been explained here: https://navid200.github.io/xDrip/docs/Nightscout/DatabaseBackup.html

Then, follow my previous instructions.

The note I show when you run bootstrap, I added because I was thinking what happens if someone runs bootstrap accidentally by mistake.  I am telling them they should not run bootstrap and I tell them what is going to happen if they proceed.
But, I am asking you to run bootstrap because I want your setup to become exactly as it should be as if you have just created a brand new setup.  So, the warning you read is to be acknowledged in this case, not to be read with surprise.  

So, if anything goes wrong, if you have created a backup and moved it off your virtual machine, you have a backup containing all of your database and all of your variables as they are right now.

**Reply by emp-00** [reply]
Ok, I just ran "Backup MongoDB and variables" - Fyi, the output of the script looks broken, not sure if this is due to my configuration or a general script issue?

![image](https://github.com/NightscoutFoundation/xDrip/assets/60714349/1a33494e-e064-45e0-9f2c-e9c01476c5fb)

**Reply by emp-00** [reply]
Ok, backup created and downloaded successfully. Now I did run "Bootstrap the stable release" from the Google Cloud setup submenu. This took only 10 seconds and then I ended up back in the "Status page" -> no forced reboot happend! I'm now rebooting manually.

**Reply by emp-00** [reply]
Reboot successful, NS instance still running as before, no configuration / data loss happened. Now executing "phase 1" again, in progress...

**Comment by emp-00** [comment]
After "phase 1" finished I rebooted the server again... Result: NS still running on 14.2.6 --> also this procedure did not change anything! I wonder what's going on? Here's the latest status page output:
![image](https://github.com/NightscoutFoundation/xDrip/assets/60714349/8288aa37-e213-4c1a-b3f6-ebcb3cd1762f)
![image](https://github.com/NightscoutFoundation/xDrip/assets/60714349/7c830e25-e38b-47b4-84b2-9c743b98b39a)
![image](https://github.com/NightscoutFoundation/xDrip/assets/60714349/e1b1086e-4462-4462-844e-a2713a63ac3e)

**Reply by Navid200** [reply]
I'm sorry about this.  I suspect that running bootstrap from the menu did not really run bootstrap!

Do you mind running it again by existing the menu and copying from the following page the link and pasting it into your virtual terminal and hitting enter?
https://navid200.github.io/xDrip/docs/Nightscout/NS_Install.html

**Reply by Navid200** [reply]
Another possibility we have not considered here is that the Google Cloud project you have been making these recent changes in is not related to the Nightscout you are looking at.  It is a possibility.  I have no way of knowing that one way or the other.  But, you can look on the status page and at the bottom, choosing option 2, to verify.

**Reply by emp-00** [reply]
Ok, I tried the "manual" bootstrap via the curl command as per your instructions, it does the same as the menu option. I made some screenshots.

![image](https://github.com/NightscoutFoundation/xDrip/assets/60714349/1649be13-1bd4-48e5-a8f4-8b4c1f2e6ea9)

then this warning message appears, I hit "ok" ...
![image](https://github.com/NightscoutFoundation/xDrip/assets/60714349/23bb0805-e13c-41b1-b454-01e0139c4bdd)

then I see this, it only takes 5 seconds only managed to shoot the first output:
![image](https://github.com/NightscoutFoundation/xDrip/assets/60714349/5119a07c-e601-4009-a997-41497a253fa8)

then I get this confirmation dialog:
![image](https://github.com/NightscoutFoundation/xDrip/assets/60714349/7de68ebc-cc9a-46a1-b3d4-2fec966ad6a6)

seeing this shortly:
![image](https://github.com/NightscoutFoundation/xDrip/assets/60714349/d4c84867-63e7-4e62-aeaf-77d6370ec239)

and ending on the status page:
![image](https://github.com/NightscoutFoundation/xDrip/assets/60714349/be80b520-2f89-4685-a361-4d6c0f028712)

**Reply by Navid200** [reply]
When you are on the status page, press 2 followed by enter.  Do you see the hostname as the site you go to to check the version?

**Comment by emp-00** [comment]
The "Login credentials" show the below output, I can 100% confirm this is my NS instance that I'm monitoring and showing 14.2.6 ... I really don't have another instance or another google cloud VM - not that I'm aware of.... Only this one with the correct url and secret;

![image](https://github.com/NightscoutFoundation/xDrip/assets/60714349/9e7b1f8f-fdc5-4927-ba1e-a2752ebe5e41)

Exactly this site (reloaded):
![image](https://github.com/NightscoutFoundation/xDrip/assets/60714349/8390df1a-806e-45bd-a804-bbabe89e8b28)

**Reply by Navid200** [reply]
Are you using FreeDNS or are you still using noip.com?

**Reply by emp-00** [reply]
Yes, I'm using freedns.afraid.org!

**Reply by emp-00** [reply]
I see my freedns domain on the login credentials page -> it's all correct, but the NS instance remains on 14.2.6 ...

**Reply by Navid200** [reply]
OK

Then, run phase 2 now.
You don't need to reenter your API_SECRET again when it asks you to.  Just press escape.
Look under Phse 2 heading here: https://navid200.github.io/xDrip/docs/Nightscout/NS_Install.html

**Reply by emp-00** [reply]
My terminal shows this with ls

![image](https://github.com/NightscoutFoundation/xDrip/assets/60714349/4aca6542-b8e5-4292-90e6-640eb9c339d6)

**Reply by Navid200** [reply]
Run phase 2 again.

**Reply by emp-00** [reply]
phase 2 completed.

![image](https://github.com/NightscoutFoundation/xDrip/assets/60714349/0b41db42-a01a-42ec-a414-65b14bee350e)
![image](https://github.com/NightscoutFoundation/xDrip/assets/60714349/98b67e34-ce7d-4889-ba22-8be8d36494aa)

HOWEVER, now my ns instance is unresponsive:
![image](https://github.com/NightscoutFoundation/xDrip/assets/60714349/1831886d-e807-4d64-b65d-eac6b7fc43ef)

**Reply by emp-00** [reply]
rebooting again ...

**Reply by emp-00** [reply]
now I see my correct freedns user ID and password here:
![image](https://github.com/NightscoutFoundation/xDrip/assets/60714349/c123c14b-0bb0-4e7c-bdbe-b2868938f989)

but my ns instance remains unrepsonsive!
![image](https://github.com/NightscoutFoundation/xDrip/assets/60714349/30a7cd11-c515-4dc0-8815-84d9b826168f)

**Reply by Navid200** [reply]
That's OK.
Give it one more minute.
If it is still nonresponsive, run phase 1 again.  This time, you won't need to run phase 2 again.

**Reply by emp-00** [reply]
I waited more than 1 minute - still nonresponsive. OK, will run phase 1 again!

**Reply by emp-00** [reply]
OK! phase 1 executed once again ... and my NS is reachable again via the web interface!
![image](https://github.com/NightscoutFoundation/xDrip/assets/60714349/d8315ec4-a143-4f70-b823-c75d8cff2212)
![image](https://github.com/NightscoutFoundation/xDrip/assets/60714349/c7abb52b-98cd-4a97-8c3a-c72803de5309)

**Reply by emp-00** [reply]
BUT! xdrip does not feed data any more :-(

Unable to do REST API Upload; Bad Gateay marking record: failed

**Reply by Navid200** [reply]
Let's go one step at a time.
Does it show 15.02 now?

**Reply by emp-00** [reply]
NS now running 15.0.2 !!! At least that was a success!
![image](https://github.com/NightscoutFoundation/xDrip/assets/60714349/25a385f3-b858-45cd-b290-68c67b239cb4)

**Reply by emp-00** [reply]
... and now updated BG data is showing in the ns instance! I think REST API works again (I changed nothing in xDrip, just waited!)

**Reply by Navid200** [reply]
OK

So, xDrip settings must not correspond to the new site.
You can update xDrip using the QR code as explained here under the heading "QR code": https://navid200.github.io/xDrip/docs/Nightscout/NS_setup.html

**Reply by Navid200** [reply]
OK
Great.  I am glad it is working again.

**Comment by emp-00** [comment]


**Reply by Navid200** [reply]
Write down the old access token.
Delete it and recreate it.

Or, even better, create a new one and use the new one in the other app.

This could also be an instance where being a little bit patient may pay off.

**Comment by emp-00** [comment]
@Navid200 - THANK YOU !! I did another server reboot and now, finally, I have NS 15.0.2 up and running, all my API access is working again and I have not lost my old configuration and all data is still there! :-) :-) :-)

We can close this topic as it looks now. Chapeau and thanks again for your time+efforts! I have no idea why we had to execute phase X etcpp multiple times, but in the end success is there :-)

**Reply by Navid200** [reply, marked as answer]
Great
